lib/filewatch/watch.rb

```diff
@@ -155,6 +188,21 @@ def subscribe(stat_interval = 1, discover_interval = 5, &block)
     end # def subscribe
 
     private
+    def expired?(stat, watched_value)
+      return false unless expiry_enabled?
+      watched_value.size == stat.size && file_expired?(stat)
+    end
+
+    def discover_expired?(stat)
+      return false unless expiry_enabled?
+      file_expired?(stat)
+    end
+
+    def file_expired?(stat)
+      Time.now.to_i > (stat.mtime.to_i + @ignore_after)
```

Comment:
I think you can subtract Time values, so maybe this makes more sense:

```
(Time.now - stat.mtime) > @ignore_after
```

I think it should work the same as your code, but avoids .to_i and compares directly to the duration (ignore_after) and might make it seem easier to read?

Alternately, a new method `file_age(stat)` that returns the `Time.now - stat.mtime` and then this method becomes `file_age(stat) > @ignore_after` 

(this is mostly trivial style concerns, so we can defer/delay resolution of this.)